src/http-api.ts

```diff
@@ -128,6 +128,29 @@ export class HttpApi {
       }
     });
 
+    this.#api.get('/:did/query', async (req, res) => {
+      const options = {} as any;
+      for (const param in req.query as object) {
+        const keys = param.split('.');
+        const lastKey = keys.pop();
+        keys.reduce((obj, key) => obj[key] = obj[key] || {}, options)[lastKey] = req.query[param]
+      }
```

Comment:
The `reduce` one liner has at least 3 WTFs for mere mortals like me. Consider adding a friendly English translation:

```suggestion
      // builds a nested object from flat keys with dot notation which may share the same parent path
      for (const param in req.query as object) {
        const keys = param.split('.');
        const lastKey = keys.pop();
        keys.reduce((obj, key) => obj[key] = obj[key] || {}, options)[lastKey] = req.query[param]
      }
```